### ISSUE TITLE
RoutingMissingException for a document that has been deleted 

### ISSUE THREAD

**Issue by stakach** [couchbaselabs/elasticsearch-transport-couchbase#130]
```
[2016-06-28 21:04:53,702][WARN ][org.eclipse.jetty.servlet.ServletHandler] /default%2f54/_bulk_docs
java.lang.RuntimeException: [default] RoutingMissingException[routing is required for [default]/[mod]/[mod_1-17]]
```

XDCR to elasticsearch seems to be getting stuck on documents that have been deleted.

Is this normal?
Blowing away the ES index and rebuilding it doesn't help and the deleted documents seem to survive couchbase backups and restore procedures.

Everything works as normal when I configure the plugin to ignore errors

**Comment by sahlex** [comment]
I found the same behavior. After having a look at the code, it's totally clear that it can not work with replication 'delete' objects.

To be able to delete child documents, the _routing meta field tells which property is to be used as parent identifier. It is configured with the  parentSelector.documentTypeParentFields configuration setting.

Unfortunately, the replication protocol does not forward the deleted document but only it's id. So it's simply not possible to read the parent id from the document.

The replication implementation does not honor the document anyways in case the deleted-flag is set (besides, it would not make sense as it's empty)!

So, to conclude: Child document deletes do not work! In my opinion, the replication feature with parent/child mapping is currently broken/not usable.

**Comment by dnault** [comment]
Quick update: we're looking into some possible fixes.

With DefaultParentSelector, we'll need to do an Elasticsearch query to load the parent ID from the child document.

With RegexParentSelector, the parent ID can be derived from the child ID, so we should be able to route the deletion request without an additional query.

**Comment by dnault** [comment]
Resolved in 4c56cf9f2fe276a3e32f5c94b838019c58db58e0

**Comment by dnault** [comment]
Fixed in version 3.0.1
